nirum_wsgi.py

```diff
@@ -46,6 +46,94 @@
 ])
 
 
+class Request(object):
+
+    def __init__(self, environ):
+        self.environ = environ
+        self._raw = None
+
+    @property
+    def method(self):
+        return self.environ['REQUEST_METHOD']
+
+    @property
+    def path(self):
+        return self.environ['PATH_INFO']
+
+    @property
+    def stream(self):
+        return self.environ['wsgi.input']
+
+    @property
+    def raw(self):
+        if self._raw is None:
+            self._raw = self.stream.read()
+        return self._raw
+
+    @property
+    def text(self):
+        return self.raw.decode('utf-8')
+
+    @property
+    def querystring(self):
+        return self.environ['QUERY_STRING']
+
+    @property
+    def args(self):
+        return {
+            key: value[0] if len(value) == 1 else value
+            for key, value in urlparse.parse_qs(self.querystring).items()
+        }
+
+    @property
+    def headers(self):
+        header_prefix = 'HTTP_'
+
+        def normalize(x):
+            header = x[len(header_prefix):]
+            return '-'.join([h[0] + h[1:].lower() for h in header.split('_')])
```

Comment:
This doesn't have to be a list comprehension; it can be a generator expression instead.